File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.py
class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        mn, mx = 0, 0
        pref, ans  = 0, 0
        for i in nums:
            pref += i
            crMx, crMn = max(pref, mx), min(pref, mn)
            ans = max(ans, abs(crMx - mn), abs(crMn - mx))
            mn, mx = crMn, crMx
        return ans

File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.java
class Solution {
    public int maxAbsoluteSum(int[] nums) {
        int mn = 0;
        int pref = 0;
        int mx = 0 ;
        for(int i : nums){
            pref += i;
            mn = Math.min(mn, pref);
            mx = Math.max(mx, pref);
        }
        return mx - mn ;
    }
}